sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5812,7 +5862,17 @@ pi_result piEventRelease(pi_event Event) {
     Event->CommandData = nullptr;
   }
   if (Event->OwnZeEvent) {
-    ZE_CALL(zeEventDestroy, (Event->ZeEvent));
+    if (!DisableEventsCaching) {
+      ZE_CALL(zeEventHostReset, (Event->ZeEvent));
+      Event->Context->addZeEventToCache(Event->ZeEvent, Event->ZeEventPool,
+                                        Event->isHostVisible(),
+                                        Event->isProfilingEnabled());
```

Comment:
Can we maybe cache the PI event instead of re-creating from L0 components?

Reply:
Good call, thanks, fixed.